projects/igniteui-angular/src/lib/grids/cell.component.ts

```diff
@@ -956,6 +1118,9 @@ export class IgxGridCellComponent implements OnInit, OnChanges, OnDestroy, CellT
 
         if (this.editable && editMode && !this.intRow.deleted) {
             if (editableCell) {
+                if (this.grid.validationTrigger === 'blur') {
+                    this.grid.tbody.nativeElement.focus({ preventScroll: true });
+                }
```

Comment:
It does become a timing issue, since validationTrigger tells the formcontrol when it can actually update the value. So until a blur happens the old value will remain in the form control (no changes will be pushed)  and in the editValue of the cell as well. Hence the updateCell api will not have the up-to-date value if a blur has not happened yet.

 I'll check if it's possible to force the form control to update it's value even if the trigger has not been called.